section11/numberformat_proposed.html

```diff
@@ -463,6 +522,143 @@ <h1>UnwrapNumberFormat( _nf_ )</h1>
+          1. Let _measureUnit_ be _numberFormat_.[[MeasureUnit]].
+          1. Let _measureUnitDisplay_ be _numberFormat_.[[MeasureUnitDisplay]].
+          1. If _patterns_.[[&lt;_measureUnit_&gt;]] is *undefined*, then
+            1. Let _measureUnit_ be `"fallback"`.
+          1. Let _patterns_ be _patterns_.[[&lt;_measureUnit_&gt;]].
+          1. Let _patterns_ be _patterns_.[[&lt;_measureUnitDisplay_&gt;]].
+        1. Else If _unitType_ is `"currency"`, then
+          1. Let _currency_ be _numberFormat_.[[Currency]].
+          1. Let _currencyDisplay_ be _numberFormat_.[[CurrencyDisplay]].
+          1. Let _currencySign_ be _numberFormat_.[[CurrencySign]].
+          1. If _patterns_.[[&lt;_currency_&gt;]] is *undefined*, then
+            1. Let _currency_ be `"fallback"`.
+          1. Let _patterns_ be _patterns_.[[&lt;_currency_&gt;]].
+          1. Let _patterns_ be _patterns_.[[&lt;_currencyDisplay_&gt;]].
+          1. Let _patterns_ be _patterns_.[[&lt;_currencySign_&gt;]].
+        1. Else,
+          1. Assert: _unitType_ is `"none"`.
+          1. Let _patterns_ be _patterns_.[[`"decimal"`]].
+        1. Let _signDisplay_ be _numberFormat_.[[SignDisplay]].
+        1. Let _patterns_ be _patterns_.[[&lt;_signDisplay_&gt;]].
+        1. Let _notation_ be _numberFormat_.[[Notation]].
+        1. Let _displayNotation_ be `"standard"`.
+        1. If _notation_ is `"scientific"` or _notation_ is `"engineering"`, then
+          1. Let _displayNotation_ be `"scientific"`.
+        1. Else If _exponent_ is not 0, then
+          1. Assert: _notation_ is `"compact"`.
+          1. Let _compactDisplay_ be _numberFormat_.[[CompactDisplay]].
+          1. If _compactDisplay_ is `"short"`, then
+            1. Let _displayNotation_ be `"compactShort"`.
+          1. Else,
+            1. Assert: _compactDisplay_ is `"long"`.
+            1. Let _displayNotation_ be `"compactLong"`.
+        1. Let _patterns_ be _patterns_.[[&lt;_displayNotation_&gt;]].
+        1. If _x_ is not *NaN* and _x_ &lt; 0 or _x_ is *-0*, then
+          1. Let _pattern_ be _patterns_.[[negativePattern]].
+        1. Else If _x_ is 0, then
+          1. Let _pattern_ be _patterns_.[[zeroPattern]].
+        1. Else,
+          1. Let _pattern_ be _patterns_.[[positivePattern]].
+        1. Return _pattern_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-computeexponent" aoid="ComputeExponent">
+      <h1>ComputeExponent ( _numberFormat_, _x_ )</h1>
+      <p>
+        The abstract operation ComputeExponent computes an exponent (power of ten) by which to scale _x_ according to the number formatting settings.  It handles cases such as 999 rounding up to 1000, requiring a different exponent.
+      </p>
+      <emu-alg>
+        1. If _x_ = 0, then
+          1. Return 0.
+        1. If _x_ &lt; 0, then
+          1. Let _x_ = -_x_.
+        1. Let _magnitude_ be the greatest integer that is not greater than the base 10 logarithm of _x_.
```

Comment:
I think we've decided in another thread that this should be something like, `Let _magnitude_ be the base 10 logarithm of _x_, rounded down to the nearest integer.`

Reply:
Will resolve when I commit against #20